src/commands/topics.ts

```diff
@@ -425,67 +482,87 @@ export async function produceMessage(content: any, topic: KafkaTopic): Promise<P
   const request: ProduceRecordRequest = {
     topic_name: topic.name,
     cluster_id: topic.clusterId,
-    ProduceRequest: produceRequest,
   };
 
   let response: ProduceResponse | undefined;
   let timestamp = new Date();
 
-  response = await recordsApi.produceRecord(request);
-  // we may get a misleading `status: 200` with a nested `error_code` in the response body
-  // ...but we may also get `error_code: 200` with a successful message
-  if (response.error_code >= 400) {
-    throw new ResponseError(
```

Comment:
After the latest sidecar updates for the CCloud produce route, we shouldn't need this kind of handling.